### PR TITLE
Fix 21205

### PR DESCRIPTION
**Pull request by dfukagaw28**
https://pear.php.net/bugs/bug.php?id=21205

### CODE REVIEW

File: Mail/mimePart.php [hunk 1]
```diff
@@ -888,17 +893,19 @@ public static function encodeHeader($name, $value, $charset = 'ISO-8859-1',
                             if ($word[0] == '"' && $word[strlen($word)-1] == '"') {
                                 // de-quote quoted-string, encoding changes
                                 // string to atom
-                                $search = array("\\\"", "\\\\");
-                                $replace = array("\"", "\\");
-                                $word = str_replace($search, $replace, $word);
                                 $word = substr($word, 1, -1);
+                                $temp = preg_replace('/\\\\([\\\\"])/', '$1', $word);
+                                $word = ($temp !== NULL) ? $temp : $word;
```
Comment (alecpl) [inline Mail/mimePart.php]: I'm not sure I like it. What's better, return an empty string on error or non-de-quoted? Anyway, preg_replace() error here would be really a rare case. I propose, to just use the return value and not check if it's null.

File: Mail/mimePart.php [hunk 2]
```diff
@@ -849,11 +849,16 @@ public static function encodeHeader($name, $value, $charset = 'ISO-8859-1',
             $charset = 'ISO-8859-1';
         }
 
+        $mb = function_exists('mb_convert_encoding');
```
Comment (alecpl) [inline Mail/mimePart.php]: Maybe check if $charset != 'UTF-8' here, so we don't call mb_convert_encoding() if not needed.

File: tests/test_Bug_21205.phpt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+require_once('Mail/mimePart.php');
+$tests = [
+    '□京都府□',
+    '∠∠∠∠',
+];
+$addr = ' <aaa@bbb.ccc>';
+$charset = 'ISO-2022-JP';
+$encoding = 'base64';
+foreach ($tests as $test) {
+    $test = mb_convert_encoding($test, $charset);
```
Comment (alecpl) [inline tests/test_Bug_21205.phpt]: In --SKIPIF-- section we should check if mbstring is available.

**Comment by dfukagaw28** [comment]
I made another commit which hopefully meet your requests, thanks.

**Pull request merged**